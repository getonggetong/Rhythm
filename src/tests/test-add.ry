/* Addition Test

	The following tests test
	the following:
	1. Simple addition
	2. Note pitch change
	3. Chord pitch change
	4. Sequence pitch change with a chord
	5. Note mixing (this makes a chord)
	6. Note mixing (with left associativity,
		this first makes a chord with A and B
		and then makes two chords with A and C
		and B and C)
	7. Chord mixing (the LHS becomes three chords
		that each include D)
	8. Chord mixing (Elements sharing an index
					will combine to make new chords)
*/

init() {
	print(1+1);
	print(A4 + 1);
	print([A4,B4] + 1);
	print([A4,B4,[C4,D4]] + 1);
	print(A4 + B4);
	print(A4 + B4 + C4);
	print([A4,B4,C4] + D4);
	print([A4,B4,C4] + [D4,E4]);
}