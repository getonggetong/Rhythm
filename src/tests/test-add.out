2
A#4.4
[A#4.4,C4.4]
[A#4.4,C4.4,[C#4.4,D#4.4]]
[A4,B4]
[[A4,C4],[B4,C4]]
[[A4,D4],[B4,D4],[C4,D4]]
[[A4,D4],[B4,E4],C4]
